apps/snitch_core/test/data/model/tax_rate_test.exs

```diff
@@ -0,0 +1,147 @@
+defmodule Snitch.Data.Model.TaxRateTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+
+  import Snitch.Factory
+
+  alias Snitch.Data.Model.TaxRate, as: TaxRateModel
+  alias Snitch.Data.Schema.TaxRate
+
+  @valid_params %{
+    name: "Europe",
+    value: 0.5,
+    included_in_price: false,
+    calculator: "DefaultCalculator"
+  }
+
+  describe "create/1" do
+    test "tax rate successfully" do
+      params = create_tax_params()
+      assert {:ok, _} = TaxRateModel.create(params)
+    end
+
+    test "fails, name missing" do
+      params = create_tax_params()
+      params = Map.drop(params, [:name])
+      assert {:error, cs} = TaxRateModel.create(params)
+      assert %{name: ["can't be blank"]} = errors_on(cs)
+    end
+
+    test "fails, calculator missing" do
+      params = create_tax_params()
+      params = Map.drop(params, [:calculator])
+      assert {:error, cs} = TaxRateModel.create(params)
+      assert %{calculator: ["can't be blank"]} = errors_on(cs)
+    end
+  end
+
+  describe "update/2" do
+    setup :tax_rate
+
+    test "successfuly calculator field", context do
+      params = %{calculator: "FlatRateCalculator"}
+      %{tax_rate: tr} = context
+      assert {:ok, tr_new} = TaxRateModel.update(params, tr)
+      assert tr_new.id == tr.id
+      refute tr_new.calculator == tr.calculator
+    end
+
+    test "successful without instance", context do
```

Comment:
Please don't test the behaviour of the `QueryHelper` here! You just have to cover one of the clauses.